crates/bevy_core_pipeline/src/core_3d/mod.rs

```diff
@@ -41,6 +41,26 @@ pub mod graph {
 // PERF: vulkan docs recommend using 24 bit depth for better performance
 pub const CORE_3D_DEPTH_FORMAT: TextureFormat = TextureFormat::Depth32Float;
 
+/// True if multisampled depth textures are supported on this platform.
+///
+/// In theory, Naga supports depth textures on WebGL 2. In practice, it doesn't,
+/// because of a silly bug whereby Naga assumes that all depth textures are
+/// `sampler2DShadow` and will cheerfully generate invalid GLSL that tries to
+/// perform non-percentage-closer-filtering with such a sampler. Therefore we
+/// disable depth of field entirely on WebGL 2.
+#[cfg(all(feature = "webgl", target_arch = "wasm32", not(feature = "webgpu")))]
+pub const DEPTH_TEXTURE_SAMPLING_SUPPORTED: bool = false;
+
+/// True if multisampled depth textures are supported on this platform.
+///
+/// In theory, Naga supports depth textures on WebGL 2. In practice, it doesn't,
+/// because of a silly bug whereby Naga assumes that all depth textures are
+/// `sampler2DShadow` and will cheerfully generate invalid GLSL that tries to
+/// perform non-percentage-closer-filtering with such a sampler. Therefore we
+/// disable depth of field entirely on WebGL 2.
```

Comment:
should this be talking about SSR instead of/as well as DOF?

Reply:
Done.